keras/activations.py

```diff
@@ -7,13 +7,9 @@ def softmax(x):
     if ndim == 2:
         return K.softmax(x)
     elif ndim == 3:
-        # apply softmax to each timestep
-        def step(x, states):
-            return K.softmax(x), []
-        last_output, outputs, states = K.rnn(step, x,
-                                             [],
-                                             mask=None)
-        return outputs
+        e = K.exp(x)
```

Comment:
Right. Thats equivalent to multiplying numerator and denominator of softmax expression by `e ^ (-max(x))`, so output will be unchanged.